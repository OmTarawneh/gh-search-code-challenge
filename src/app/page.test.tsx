import { render } from "@testing-library/react";
import Home from "./page";

// Mock the components used in Home
jest.mock("../components/Banner", () => () => <div>Banner Component</div>);
jest.mock("../components/SearchResult", () => () => (
	<div>SearchResult Component</div>
));
jest.mock("../components/Search", () => () => (
	<div>SearchSection Component</div>
));

describe("Home", () => {
	it("renders all the required components", () => {
		const { container } = render(<Home />);

		// Ensure that the required components are rendered
		expect(container.textContent).toContain("Banner Component");
		expect(container.textContent).toContain("SearchResult Component");
		expect(container.textContent).toContain("SearchSection Component");
	});

	it("matches the snapshot", () => {
		const { container } = render(<Home />);

		// Check if the rendered component matches the snapshot
		expect(container).toMatchSnapshot();
	});
});
