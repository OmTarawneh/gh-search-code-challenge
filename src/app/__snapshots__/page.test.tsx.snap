// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches the snapshot 1`] = `
<div>
  <div
    class="MuiStack-root css-1reytsa-MuiStack-root"
  >
    <div>
      Banner Component
    </div>
    <div>
      SearchSection Component
    </div>
    <div>
      SearchResult Component
    </div>
  </div>
</div>
`;
